Xamarin.Forms.Xaml/MarkupExtensions/OnPlatformExtension.cs

```diff
@@ -22,21 +22,34 @@ public class OnPlatformExtension : IMarkupExtension
 
 		public object ProvideValue(IServiceProvider serviceProvider)
 		{
-			var lineInfo = serviceProvider?.GetService<IXmlLineInfoProvider>()?.XmlLineInfo;
-			if (Android == null && GTK == null && iOS == null && 
-				macOS == null && Tizen == null && UWP == null && 
-				WPF == null && Default == null)
-			{
-				throw new XamlParseException("OnPlatformExtension requires a non-null value to be specified for at least one platform or Default.", lineInfo ?? new XmlLineInfo());
+			if (   Android == null
```

Comment:
Humbly suggest hoisting this logic into a property `Platform` that returns the platform so this callsite could test `Property == null`.